api/fixtures/example.go

```diff
@@ -226,23 +196,100 @@ aws_secret_access_key = %s
 				PublicZoneID:  o.PublicZoneID,
 				PrivateZoneID: o.PrivateZoneID,
 			},
-			Platform: hyperv1.PlatformSpec{
-				Type: hyperv1.AWSPlatform,
-				AWS: &hyperv1.AWSPlatformSpec{
-					Region: o.AWS.Region,
-					Roles:  o.AWS.Roles,
-					CloudProviderConfig: &hyperv1.AWSCloudProviderConfig{
-						VPC: o.AWS.VPCID,
-						Subnet: &hyperv1.AWSResourceReference{
-							ID: &o.AWS.SubnetID,
-						},
-						Zone: o.AWS.Zone,
+			Platform: hyperv1.PlatformSpec{},
+		},
+	}
+
+	switch o.PlatformType {
+	case "aws":
+		cluster.Spec.Platform.Type = hyperv1.AWSPlatform
+		cluster.Spec.SigningKey = corev1.LocalObjectReference{Name: signingKeySecret.Name}
+		cluster.Spec.Platform.AWS = &hyperv1.AWSPlatformSpec{
+			Region: o.AWS.Region,
+			Roles:  o.AWS.Roles,
+			CloudProviderConfig: &hyperv1.AWSCloudProviderConfig{
+				VPC: o.AWS.VPCID,
+				Subnet: &hyperv1.AWSResourceReference{
+					ID: &o.AWS.SubnetID,
+				},
+				Zone: o.AWS.Zone,
+			},
+			KubeCloudControllerCreds: corev1.LocalObjectReference{Name: kubeCloudControllerCredsSecret.Name},
+			NodePoolManagementCreds:  corev1.LocalObjectReference{Name: nodePoolManagementCredsSecret.Name},
+		}
+		cluster.Spec.Services = []hyperv1.ServicePublishingStrategyMapping{
+			{
+				Service: hyperv1.APIServer,
+				ServicePublishingStrategy: hyperv1.ServicePublishingStrategy{
+					Type: hyperv1.LoadBalancer,
+				},
+			},
+			{
+				Service: hyperv1.OAuthServer,
+				ServicePublishingStrategy: hyperv1.ServicePublishingStrategy{
+					Type: hyperv1.Route,
+				},
+			},
+			{
+				Service: hyperv1.OIDC,
+				ServicePublishingStrategy: hyperv1.ServicePublishingStrategy{
+					Type: hyperv1.Route,
+				},
+			},
+			{
+				Service: hyperv1.Konnectivity,
+				ServicePublishingStrategy: hyperv1.ServicePublishingStrategy{
+					Type: hyperv1.LoadBalancer,
+				},
+			},
+			{
+				Service: hyperv1.Ignition,
+				ServicePublishingStrategy: hyperv1.ServicePublishingStrategy{
+					Type: hyperv1.Route,
+				},
+			},
+		}
+
+	case "none":
+		cluster.Spec.Platform.Type = hyperv1.NonePlatform
+		cluster.Spec.Services = []hyperv1.ServicePublishingStrategyMapping{
+			{
+				Service: hyperv1.APIServer,
+				ServicePublishingStrategy: hyperv1.ServicePublishingStrategy{
+					Type: hyperv1.NodePort,
+					NodePort: &hyperv1.NodePortPublishingStrategy{
+						Address: o.Annotations["master_0.Address"],
```

Comment:
Could we take this as a `none` only CLI flag rather than a bespoke annotation

Reply:
Sorry, but I don't understand it :( My idea is just to use the k8s client to get the master0.Address instead of asking/retrieving the information with the flag. With master_0 you fit for SNO/Multinode guest cluster.

Reply:
Yeah, my comment wasn't clear.

I was thinking about taking this as a flag on `create cluster` like `--external-api-server-address` that is only used if `--platform=none` is also set.

Reply:
It could be... but why do you need it? I mean, if you have the information for the master address for guest cluster, why should be added by an extra flag? Remember that this information is only for a NodePort Address until Route will be developed/available. That's why I decided to use the master0 Ip address until 'Route' will be available (the same to Konnectivity as you did in the other PR).

Reply:
Then again... do we? I thought we originally had discussed constraining the scope of the CLI to the bare minimum necessary to support the "kick the tires" or demonstrative use case with AWS (and to provide some baseline smoke test for CI).

For everything else, I thought we had decided users should be constructing their manifests using their own more sophisticated tooling.

I'd like to know how we can draw a more clear line with regards to the intended use cases for the CLI so that we can define and avoid scope creep.

Reply:
@ironcladlou My issue is that one can't easily get a HostedCluster/NodePool spec out of the CLI without side effects.

Right now, I run the CLI with `--render > cluster.yaml` which requires AWS creds and creates a ton of resources in AWS.  Then I immediately run the `destroy cluster` command to delete them and make the manual modification to `cluster.yaml` for platform `None` before doing an `oc create -f cluster.yaml`.

Our current unstable API means there are no such higher level tools and we are left to explain repetitively how to convert an platform `AWS` HostedCluster/NodePool to a platform `None`.

I do think the CLI should have some way to get a basic HostedCluster/NodePool manifest out with no side effects.

Reply:
Makes sense... I like the idea of making these quality of life improvements. I do wonder in a bigger picture way where else we could be focusing on UX (CLI commands, templates in $tempate_system, static examples, API documentation, example/guide type documentation) generally, but we can take that conversation elsewhere.

Thanks for working on this!

Reply:
> Yeah, my comment wasn't clear.
> 
> I was thinking about taking this as a flag on `create cluster` like `--external-api-server-address` that is only used if `--platform=none` is also set.

changed.